crates/flowctl/src/raw.rs

```diff
@@ -123,12 +141,77 @@ async fn do_rpc(
     Ok(())
 }
 
-pub async fn do_bundle(
+async fn do_bundle(_ctx: &mut crate::CliContext, Bundle { source }: &Bundle) -> anyhow::Result<()> {
+    let catalog = crate::source::bundle(source).await?;
+    serde_json::to_writer_pretty(io::stdout(), &catalog)?;
+    Ok(())
+}
+
+async fn do_combine(
     _ctx: &mut crate::CliContext,
-    Bundle { source }: &Bundle,
+    Combine { source, collection }: &Combine,
 ) -> anyhow::Result<()> {
+    let collection = models::Collection::new(collection);
     let catalog = crate::source::bundle(source).await?;
-    serde_json::to_writer_pretty(std::io::stdout(), &catalog)?;
+
+    let Some(models::CollectionDef{schema: models::Schema::Object(schema), key, .. }) = catalog.collections.get(&collection) else {
```

Comment:
Just as an aside, I'm kinda thinking that `flowctl` ought to have a general way of handling references to entities that allows users to either refer to the definition in a local file (like it is here), or refer to the entity by name and have `flowctl` fetch it from the control plane. Totally cool to leave this code as it is right now, but just checking if we're aligned on the vision for `flowctl` UX.

Reply:
👍